### DISCUSSION TITLE
FileUplink/Downlink CFDP Serialization

### DISCUSSION THREAD

**Discussion by aditsachde** [nasa/fprime#1650] [Q&A]
Hi! We need to implement file uplink and downlink functionality in a way very similar to what the `FileUplink` and `FileDownlink` components provide. However, we are required to conform to the CCSDS/CFDP specifications.

Looking through the components, they use `Fw::FilePacket`, which seems to be modeled after the CFDP specification, but isn't serialized in the same way. Would it be a viable to write a component that translates between the `Fw::FilePacket` wire format and the CCSDS/CFDP wire format?

**Comment by timcanham** [comment]
Certainly! That's the F Prime tinkertoy way. We would definitely be interested in how you did it.

**Comment by rrieber** [comment]
@aditsachde Did you ever get CFDP incorporated into F-Prime? I'm exploring the same problem and curious how difficult it was for you.

For future readers of this discussion, I found some open repositories for CFDP that may or may not be useful:
 - https://github.com/nasa/CF
 - https://sourceforge.net/p/ion-dtn/code/ci/current/tree/cfdp/
